File: algorithms/378. Kth Smallest Element in a Sorted Matrix/solution.cc
class Solution {
public:
    int kthSmallest(vector<vector<int>>& matrix, int k) {
        vector<int> flat;
        for ( int i = 0; i < matrix.size(); ++i ) {
            flat.insert(flat.end(), matrix[i].begin(), matrix[i].end());
        }
        sort(flat.begin(), flat.end());
        return flat[k-1];
    }
};

// 85 / 85 test cases passed.
// Status: Accepted
// Runtime: 53 ms
// beats 38.20 %

File: algorithms/378. Kth Smallest Element in a Sorted Matrix/solution.py
class Solution(object):
    def kthSmallest(self, matrix, k):
        """
        :type matrix: List[List[int]]
        :type k: int
        :rtype: int
        """
        # return list(heapq.merge(*matrix))[k-1]
        return sorted(itertools.chain(*matrix))[k-1]

# 85 / 85 test cases passed.
# Status: Accepted
# Runtime: 72 ms
# beats 85.09 %
